src/dune_rules/install_rules.ml

```diff
@@ -43,6 +43,27 @@ module Package_paths = struct
     Path.Build.extend_basename (meta_file ctx pkg) ~suffix:".template"
 end
 
+let check_runtime_deps_relative_path path ~loc ~lib_info =
+  let lib_src_dir = Lib_info.src_dir lib_info in
+  match
+    Path.Local.descendant
+      (path |> Path.as_in_build_dir_exn |> Path.Build.local)
```

Comment:
How about have the caller do this cast instead?

Reply:
How would you suggest getting back the original path to return from this function? (currently line 65)

Alternative I can think of:
- make this function return `unit` and just check the path?

Reply:
> make this function return unit and just check the path?

I think that would be even better.

Reply:
Ok I pushed that alternative.